src/interpreter/plugin/FinancialPlugin.ts

```diff
@@ -62,29 +223,372 @@ export class FinancialPlugin extends FunctionPlugin {
+      (pmtCore(rate, periods, present, future, type) - ipmtCore(rate, period, periods, present, future, type))
+    )
   }
 
   public fv(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
     return this.runFunction(ast.args, formulaAddress, this.metadata('FV'), fvCore)
   }
+
+  public cumipmt(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('CUMIPMT'),
+      (rate: number, periods: number, value: number, start: number, end: number, type: number) => {
+        if (start > end) {
+          return new CellError(ErrorType.NUM)
+        }
+
+        const payment = pmtCore(rate, periods, value, 0, type)
+        let acc = 0
+
+        if (start === 1) {
+          if (type === 0) {
+            acc = -value
+          }
+          start = 2
+        }
+
+        for (let i = start; i <= end; i++) {
+          acc += type ? fvCore(rate, i - 2, payment, value, type) - payment : fvCore(rate, i - 1, payment, value, type)
+        }
+        return acc * rate
+      }
+    )
+  }
+
+  public cumprinc(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('CUMPRINC'),
+      (rate: number, periods: number, value: number, start: number, end: number, type: number) => {
+        if (start > end) {
+          return new CellError(ErrorType.NUM)
+        }
+
+        const payment = pmtCore(rate, periods, value, 0, type)
+        let acc = 0
+        if (start === 1) {
+          acc = type ? payment : payment + value*rate
+          start = 2
+        }
+        for (let i = start; i <= end; i++) {
+          acc += payment - rate * (type ? (fvCore(rate, i - 2, payment, value, 1) - payment) : fvCore(rate, i - 1, payment, value, 0))
+        }
+
+        return acc
+      }
+    )
+  }
+
+  public db(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
+    return this.runFunction(ast.args, formulaAddress, this.metadata('DB'),
+      (cost: number, salvage: number, life: number, period: number, month: number) => {
+        if ((month===12 && period > life) || (period > life+1)) {
+          return new CellError(ErrorType.NUM)
+        }
+
+        if (salvage >= cost) {
+          return 0
+        }
+
+        const rate = Math.round((1 - Math.pow(salvage / cost, 1 / life))*1000)/1000
+
+        const initial = cost * rate * month / 12
+
+        if(period===1) {
+          return initial
+        }
+
+        let total = initial
+
+        for (let i = 0; i < period - 2; i++) {
+          total += (cost - total) * rate
+        }
+        if(period === life+1) {
+          return (cost - total) * rate * (12-month)/12
+        }
+        return (cost - total) * rate
+      }
+    )
+  }
+
+  public ddb(ast: ProcedureAst, formulaAddress: SimpleCellAddress): InternalScalarValue {
```

Comment:
shouldn't we allow also non-integer `period`s?

Reply:
sure, it should be easy